python/sglang/srt/layers/quantization/quark/schemes/quark_w4a4_mxfp4.py

```diff
@@ -0,0 +1,105 @@
+# SPDX-License-Identifier: Apache-2.0
```

Comment:
Design logic should be (with mxfp4 weight):
- On HW does not support mxfp matmul, we need to use matmul (and activation dtype) of fp16/bf16 or fp8, instructed by `SGLANG_QUARK_MXFP_DQ16` or `SGLANG_QUARK_MXFP_DQ8`, remove the unclear name `SGLANG_QUARK_EMU_MEM_OPT`
- On HW support mxfp matmal, quantize activation to mxfp, and call mxfp matmul (this is separate from this PR)

Reply:
Yea quark would not dictate the kernel to be used. The current implementation allows end-to-end runs and validation for mxfp4-mxfp4 gemm on HW that does not support it. The flag is controlling a trade-off between better perf and lower memory usage. The flag can be dropped if you feel it's not making sense.

fp16/bf16 or fp8 kernels can be follow-ups but I'm not sure how important they are, unless there are efficient fp8/fp16-mxfp4 kernels and a use case for them?